drv/transceivers-server/src/main.rs

```diff
@@ -253,28 +255,24 @@ impl ServerImpl {
 
         #[derive(Copy, Clone, FromBytes, AsBytes)]
         #[repr(C)]
-        struct StatusAndTemperature {
-            status: u8,
+        struct Temperature {
             temperature: zerocopy::I16<zerocopy::BigEndian>,
         }
 
-        loop {
-            let mut out = StatusAndTemperature::new_zeroed();
-            self.transceivers
-                .get_i2c_status_and_read_buffer(port, out.as_bytes_mut())?;
-            if out.status & Reg::QSFP::PORT0_STATUS::BUSY == 0 {
-                if out.status & Reg::QSFP::PORT0_STATUS::ERROR != 0 {
-                    return Err(FpgaError::ImplError(out.status));
-                } else {
-                    // "Internally measured free side device temperatures are
-                    // represented as a 16-bit signed twos complement value in
-                    // increments of 1/256 degrees Celsius"
-                    //
-                    // - SFF-8636 rev 2.10a, Section 6.2.4
-                    return Ok(Celsius(out.temperature.get() as f32 / 256.0));
-                }
-            }
-            userlib::hl::sleep_for(1);
+        let mut out = Temperature::new_zeroed();
+        let status = self
+            .transceivers
+            .get_i2c_status_and_read_buffer(port, out.as_bytes_mut())?;
+
+        if status.error == FpgaI2CFailure::NoError {
+            // "Internally measured free side device temperatures are
+            // represented as a 16-bit signed twos complement value in
+            // increments of 1/256 degrees Celsius"
+            //
+            // - SFF-8636 rev 2.10a, Section 6.2.4
+            Ok(Celsius(out.temperature.get() as f32 / 256.0))
+        } else {
+            Err(FpgaError::ImplError(status.as_bytes()[0]))
```

Comment:
What is this doing?  I think it's going to return the value of `rdata_fifo_empty`, because it's casting an `PortI2CStatus` to a `&[u8]` and taking the first byte.

Reply:
Good catch! I actually don't care about the whole status here, I really just want to log the `error` observed. I'm going to make this line:
```suggestion
            Err(FpgaError::ImplError(status.error.as_bytes()[0]))
```
And get rid of the `AsBytes` deriviation on the `PortI2CStatus` type.

Reply:
One more note: it looks like you can do `status.error as u8` instead of `status.error.as_bytes()[0]`, since it's already a single-byte `enum` type.